text/0000-slatepack.md

```diff
@@ -0,0 +1,391 @@
+
+### Without a `SlatePack Address`
+
+1. `grin-wallet send 1.337`
+
+2. A `SlatePack Message` string is output for manual asynchronous transport by the user
+
+### With QR Codes
+
+A QR-based `SlatePack Workflow` will always begin with a standard QR size because they are encoded directly from a bech32 `SlatePack Address`.
+
+This encoding simultaneously provides a derivable onion address to attempt a synchronous transaction (`bech32 -> ed25519 -> onionv3`) _and_ a derivable encryption key (`bech32 -> ed25519 -> x25519`) to return an encrypted SlatePack string to complete the transaction asynchronously as a fallback.
+
+As a consequence, a `SlatePack` address must be revealed by the party producing a QR code in the `SlatePack Workflow`.
+
+1. Receiver shares `SlatePack Address` via QR
+
+2. Sender scans QR code and the transaction is completed synchronously via Tor by deriving the recipients onion v3 address from their `SlatePack Address`
+
+3. (Fallback) If the synchronous transaction fails, a `SlatePack Message` string is output for manual asynchronous transport by the user
+
+## Implementation Timeline
+
+1. Initial SlatePack implementation introduced with the _July 2020 hard fork_
+  - May or may not support encryption by default yet
+
+2. The proposed SlatePack standard is fully implemented and adopted as a universal transaction standard in _last hard fork (Jan 2021)_
+  - SlatePack is the default transaction standard in all wallets and services
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This puts a lot of eggs in one basket (if SlatePack fails there will likely be confusion returning to old methods)
+
+- This may be a bit rushed to have where we want it before HF schedule
+
+- Deprecating HTTP(S) is already a major change- by requiring the adoption of this completely new standard in addition we risk putting a lot of effort on the shoulders of existing services in the Grin ecosystem
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- By adding new options without simplifying the workflow for users we risk confusion and friction
+
+- We could just add an option for copy and pastable slates instead of introducing an entirely new universal transaction flow standard
+
+# Prior art
+[prior-art]: #prior-art
+
+## Tari
+
+Tari handles this general problem by...
+
+## Beam
+
+Beam handles this problem by...
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- How to handle key derivation harmoniously?
+
+- What are unmentioned security considerations for using the same base key to both map to an onion address and map to an encryption key used in transactions?
+  - Related, what are unmentioned security considerations to `SlatePack Address` reuse?
+
+- What should the `human-readable part` of the bech32 encoded `SlatePack Address` be?
```

Comment:
The nice thing about bech32 is that it can be encoded to QR very efficiently. It also shouldn't break when copy/pasting across different applications/platforms. I'll add this as an alternative option though as it seems relevant today.

Reply:
Sorry, I was trolling. You probably missed my rants about Tari's emoji addresses in keybase :).